File: androidApp/src/main/kotlin/subha/app/cyra/ui/profilesetup/ProfileSetupWeightStep.kt
package subha.app.cyra.ui.profilesetup

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.width
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import subha.app.cyra.R
import subha.app.cyra.feature.profilesetup.domain.WeightConverter
import subha.app.cyra.feature.profilesetup.domain.WeightUnit
import subha.app.cyra.ui.components.CyraLabeledSlider
import subha.app.cyra.ui.components.CyraSegmentedToggle
import subha.app.cyra.ui.components.CyraTextField
import subha.app.cyra.ui.components.ScaleIcon

private const val MIN_WEIGHT_KG = 30
private const val MAX_WEIGHT_KG = 150

@Composable
fun ProfileSetupWeightStep(
    weightKg: Int,
    weightUnit: WeightUnit,
    onWeightChange: (Int) -> Unit,
    onWeightUnitChange: (WeightUnit) -> Unit,
    modifier: Modifier = Modifier,
) {
    ProfileSetupStepScaffold(
        title = stringResource(R.string.profile_setup_weight_title),
        subtitle = stringResource(R.string.profile_setup_weight_subtitle),
        modifier = modifier,
    ) {
        Text(text = stringResource(R.string.profile_setup_weight_label), style = MaterialTheme.typography.titleSmall, color = MaterialTheme.colorScheme.onSurface)
        Spacer(modifier = Modifier.height(8.dp))

        val isMetric = weightUnit == WeightUnit.KG
        val displayValue = if (isMetric) weightKg else WeightConverter.kgToLb(weightKg)
        val displayLabel = if (isMetric) "$weightKg kg" else "$displayValue lb"
        val range = if (isMetric) MIN_WEIGHT_KG..MAX_WEIGHT_KG else WeightConverter.kgToLb(MIN_WEIGHT_KG)..WeightConverter.kgToLb(MAX_WEIGHT_KG)
        val ticks = if (isMetric) {
            listOf(30, 60, 90, 120, 150)
        } else {
            listOf(30, 60, 90, 120, 150).map { WeightConverter.kgToLb(it) }
        }

        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(10.dp)) {
            CyraTextField(
                value = displayValue.toString(),
                onValueChange = { input ->
                    val parsed = input.toIntOrNull() ?: return@CyraTextField
                    onWeightChange(if (isMetric) parsed else WeightConverter.lbToKg(parsed))
                },
                placeholder = stringResource(R.string.profile_setup_weight_placeholder),
                leadingIcon = { ScaleIcon() },
                keyboardType = KeyboardType.Number,
                modifier = Modifier.weight(1f),
            )
            CyraSegmentedToggle(
                options = listOf("kg", "lb"),
                selectedIndex = if (isMetric) 0 else 1,
                onOptionSelected = { index -> onWeightUnitChange(if (index == 0) WeightUnit.KG else WeightUnit.LB) },
                modifier = Modifier.width(140.dp),
            )
        }

        Spacer(modifier = Modifier.height(20.dp))

        CyraLabeledSlider(
            value = displayValue,
            range = range,
            valueLabel = displayLabel,
            ticks = ticks,
            onValueChange = { raw -> onWeightChange(if (isMetric) raw else WeightConverter.lbToKg(raw)) },
        )
    }
}

File: iosApp/iosApp/ProfileSetup/ProfileSetupWeightStep.swift
import SwiftUI
import SharedLogic

private let minWeightKg: Int32 = 30
private let maxWeightKg: Int32 = 150

struct ProfileSetupWeightStep: View {
    let weightKg: Int32
    let weightUnit: WeightUnit
    let onWeightChange: (Int32) -> Void
    let onWeightUnitChange: (WeightUnit) -> Void

    var body: some View {
        ProfileSetupStepScaffold(
            title: String(localized: "profile_setup_weight_title"),
            subtitle: String(localized: "profile_setup_weight_subtitle"),
        ) {
            Text(String(localized: "profile_setup_weight_label"))
                .font(CyraFont.titleSmall())
                .foregroundColor(.cyraOnSurface)
            Spacer().frame(height: 8)

            let isMetric = weightUnit == .kg
            let displayValue = isMetric ? weightKg : WeightConverter.shared.kgToLb(kg: weightKg)
            let displayLabel = isMetric ? "\(weightKg) kg" : "\(displayValue) lb"
            let range: ClosedRange<Int> = isMetric
                ? Int(minWeightKg)...Int(maxWeightKg)
                : Int(WeightConverter.shared.kgToLb(kg: minWeightKg))...Int(WeightConverter.shared.kgToLb(kg: maxWeightKg))
            let metricTicks: [Int32] = [30, 60, 90, 120, 150]
            let ticks = isMetric ? metricTicks.map { Int($0) } : metricTicks.map { Int(WeightConverter.shared.kgToLb(kg: $0)) }

            HStack(spacing: 10) {
                CyraTextField(
                    placeholder: String(localized: "profile_setup_weight_placeholder"),
                    text: Binding(
                        get: { "\(displayValue)" },
                        set: { input in
                            guard let parsed = Int32(input) else { return }
                            onWeightChange(isMetric ? parsed : WeightConverter.shared.lbToKg(lb: parsed))
                        },
                    ),
                    systemImage: "scalemass",
                    keyboardType: .numberPad,
                )
                CyraSegmentedToggle(
                    options: ["kg", "lb"],
                    selectedIndex: isMetric ? 0 : 1,
                    onOptionSelected: { index in onWeightUnitChange(index == 0 ? .kg : .lb) },
                )
                .frame(width: 140)
            }

            Spacer().frame(height: 20)

            CyraLabeledSlider(
                value: Int(displayValue),
                range: range,
                valueLabel: displayLabel,
                ticks: ticks,
                onValueChange: { raw in
                    onWeightChange(isMetric ? Int32(raw) : WeightConverter.shared.lbToKg(lb: Int32(raw)))
                },
            )
        }
    }
}
